### PR TITLE
Issue #1235 - error handling for generic error messages in 3 forms

### LINKED ISSUE #1235: Update error message new generic error messages
### Issue description

The registry can produce a connection error (no error code) or various errors that essentially mean a user entered bad data of some kind. All user events that can result in EPP errors should display plain language error messages. The content for the error messages is defined in [this word doc ](https://docs.google.com/document/d/12qxl6evZw16dT-TB9iu3YLjKdwwL5KqIYQJV4uncAXg/edit#heading=h.30j0zll)

### Acceptance criteria

- [ ] client hold/revert client hold/ delete buttons  on django admin display connection error should a connection error occur
- [ ] in the Manage Domain section, the following pages: security email and DNSSEC should show the connection if a connection error occurs. Note name servers is already implemented to do this and should be a good example for the other two pages.
- [ ]  in the Manage Domain section, the follow pages:security email, DNSSEC, and Nameservers  should show their corresponding generic error message if any other EPP error (or ContactError for security email page) is thrown. 

### Additional context

_No response_

### Links to other issues

Relates to #433

### PR DESCRIPTION
**Pull request by dave-kennedy-ecs**
## Ticket

Resolves #1235 

## Changes

- Generic error messages updated for Nameserver, DS Data and Security Email forms.
- Nameserver message is now: There’s something wrong with the name server information you provided. If you need help email us at help@get.gov.  
- DS Data message is now: There’s something wrong with the DS data you provided. If you need help email us at help@get.gov.
- Security Email message is now: Enter an email address in the required format, like name@example.com.

## Context for reviewers

Can currently be tested in [BL Sandbox](https://getgov-bl.app.cloud.gov/)

It is a bit difficult to produce these generic messages because they only occur in edge cases where there is either bad data that is not accounted for in form and field level error handling already, or when there is an error from EPP that is not a connection error.

Not certain how to reproduce generic DNSSEC DS data message, as known error conditions are all caught

Not certain how to reproduce nameserver message, as known error conditions are all caught

To reproduce security email message, go to security email form on a domain management screen.  Enter an invalid email address.

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [x] Pulled this branch locally and tested it
- [x] Reviewed this code and left comments
- [x] Checked that all code is adequately covered by tests
- [x] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [x] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [x] Checked keyboard navigability
- [x] Meets all designs and user flows provided by design/product
- [x] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [x] Chrome
  - [ ] Microsoft Edge
  - [x] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

File: src/registrar/utility/errors.py
Comment (zandercymatics) [inline src/registrar/utility/errors.py:159]:
```suggestion
        - 1 BAD_DATA  (bad data input in security email)
```
(nitpick/optional)  I think the name + the description of it should have some sort of separator. Could be a `:` for instance. Makes it slightly easier to read but a super minor thing

**Review by zandercymatics — COMMENTED** [review]
I can't directly test these errors given the EPP connection issue, but I did run your branch locally and it worked fine. Seemed OK when I manually invoked them, as well. 

Code looks great. Will pass through this again once we get that and get this approved after I test with that. (unless this is MVP, in which case I can approve)

**Comment by abroddrick** [comment]
@dave-kennedy-ecs doesn't this resolve https://github.com/cisagov/manage.get.gov/issues/1235 ? 433 is just a design/content ticket which is already closed

**Review by zandercymatics — APPROVED** [review]
LGTM. Works on both Firefox and Chrome and ANDI doesn't raise anything. Approving.

Only thing worth noting here is that these errors don't really get triggered often in the supporting PR, as form validation tends to catch the invalid input before it can post. This means that we still get the generic error for most. That said, ticket #1235 specifically mentions EPP errors so I am unsure

**Review by Katherine-Osos — APPROVED** [review]

**Pull request merged**
